src/test/java/com/thomashan/coup/turn/state/WaitingChallengeMainActionStateTest.java

```diff
@@ -0,0 +1,27 @@
+package com.thomashan.coup.turn.state;
+
+import com.thomashan.coup.action.ChallengeAction;
+import com.thomashan.coup.action.ChallengeActionType;
+import org.junit.jupiter.api.Test;
+
+import static com.thomashan.coup.PlayerBuilder.newBuilder;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class WaitingChallengeMainActionStateTest extends WaitingChallengeMainActionStateTestCase {
+    @Test
+    public void testPerformAction_GivenMainActionIsNotChallengeable_ThrowsException() {
+        Throwable throwable = assertThrows(IllegalArgumentException.class, () ->
+                createWaitingChallengeMainActionState(anyNonChallengeableMainAction()).performAction(ChallengeAction.of(newBuilder().build(), ChallengeActionType.CHALLENGE)));
+        assertEquals("Challenging a non-challengeable action", throwable.getMessage());
+    }
+
+    @Test
+    public void testPerformAction_GivenMainActionIsChallengeableTurnPlayerSameAsChallenger_ThrowsException() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unit tests should not contain more than 1 assert(s).](https://app.codacy.com/app/thomashan/coup-java/pullRequest?prid=4561385)